src/packagedcode/windows_helpers.py

```diff
@@ -0,0 +1,311 @@
+import os
+from pathlib import PureWindowsPath
+
+import attr
+import logbook
+from regipy.exceptions import NoRegistrySubkeysException
+from regipy.exceptions import RegistryKeyNotFoundException
+from regipy.registry import RegistryHive
+
+from commoncode.command import execute
+from packagedcode import models
+
+
+logger = logbook.Logger(__name__)
+TRACE = False
+
+
+MSIINFO_BIN_LOCATION = 'packagedcode_msitools.msiinfo'
+
+
+def get_msiinfo_bin_location():
+    """
+    Return the binary location for msiinfo
+    """
+    from plugincode.location_provider import get_location
+    msiinfo_bin_loc = get_location(MSIINFO_BIN_LOCATION)
+    if not msiinfo_bin_loc:
+        raise Exception(
+            'CRITICAL: msiinfo not provided. '
+            'Unable to continue: you need to install the plugin packagedcode-msitools'
+        )
+    return msiinfo_bin_loc
+
+
+class MsiinfoException(Exception):
+    pass
+
+
+def parse_msiinfo_suminfo_output(output_string):
+    """
+    Return a dictionary containing information from the output of `msiinfo suminfo`
+    """
+    # Split lines by newline and place lines into a list
+    output_list = output_string.split('\n')
+    results = {}
+    # Partition lines by the leftmost ":", use the string to the left of ":" as
+    # the key and use the string to the right of ":" as the value
+    for output in output_list:
+        key, _, value = output.partition(':')
+        if key:
+            results[key] = value.strip()
+    return results
+
+
+def get_msi_info(location):
+    """
+    Run the command `msiinfo suminfo` on the file at `location` and return the
+    results in a dictionary
+
+    This function requires the `packagedcode-msiinfo` plugin to be installed on the system
+    """
+    rc, stdout, stderr = execute(
+        cmd_loc=get_msiinfo_bin_location(),
+        args=[
+            'suminfo',
+            location,
+        ],
+    )
+    if stderr:
+        error_message = f'Error encountered when reading MSI information from {location}: '
+        error_message = error_message + stderr
+        raise MsiinfoException(error_message)
+    return parse_msiinfo_suminfo_output(stdout)
+
+
+def msi_parse(location):
+    """
+    TODO: get proper package name and version from MSI
```

Comment:
Do you think you could add this before we merge?

Reply:
BTW if you ever need to extract some version from a name, this may help https://github.com/nexB/commoncode/blob/452c2b513231460d5fa97bccb0acfafa16356682/src/commoncode/version.py
so may libversion from repology?

Reply:
Thanks. I modified the code there to get version number from the `subject` line of the `msiinfo` results here: https://github.com/nexB/scancode-toolkit/blob/cab062d14f06273bc173b651e57ebb95ffa6aa32/src/packagedcode/msi.py#L148